### PR TITLE
feat(onyx-1219): add Collections home view rail

### PR DESCRIPTION
**Pull request by nickskalkin**
This PR resolves [ONYX-1219]

### Description

Do not merge until https://github.com/artsy/metaphysics/pull/5934 is merged!

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#nochangelog

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1219]: https://artsyproduct.atlassian.net/browse/ONYX-1219?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/HomeView/Sections/MarketingCollectionRailItem.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  const marketingCollection = useFragment(fragment, marketingCollectionFragment)
+
+  // Collections are expected to always have >= 2 artworks, but we should
+  // still be cautious to avoid crashes if this assumption is broken.
+  const artworkImageURLs = compact(
+    extractNodes(marketingCollection.artworksConnection, (artwork) => artwork.image?.url)
+  )
+
+  return (
+    // eslint-disable-next-line react/jsx-no-undef
```
Comment (MounirDhahri) [inline src/app/Scenes/HomeView/Sections/MarketingCollectionRailItem.tsx:28]: is this still needed?

File: src/app/Scenes/HomeView/Sections/MarketingCollectionRailItem.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      <Flex>
+        <FiveUpImageLayout imageURLs={artworkImageURLs} />
+
+        <Flex mt={1}>
+          <Text variant="sm-display" numberOfLines={1} weight="medium">
+            {marketingCollection?.title}
+          </Text>
+          <Text variant="xs" numberOfLines={1} color="black60">
+            {marketingCollection?.artworksConnection?.counts?.total
+              ? `${marketingCollection.artworksConnection.counts.total} works`
```
Comment (MounirDhahri) [inline src/app/Scenes/HomeView/Sections/MarketingCollectionRailItem.tsx:48]: We can use `pluralize()` helper here

File: src/app/Scenes/HomeView/Sections/MarketingCollectionsRailHomeViewSection.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  section: MarketingCollectionsRailHomeViewSection_section$key
+}
+
+export const MarketingCollectionsRailHomeViewSection: React.FC<
+  MarketingCollectionsRailHomeViewSectionProps
+> = ({ section }) => {
+  const data = useFragment(fragment, section)
+  const component = data.component
+
+  if (!component) return null
```
Comment (MounirDhahri) [inline src/app/Scenes/HomeView/Sections/MarketingCollectionsRailHomeViewSection.tsx:19]: I really think we should make `component` non nullable. So far all rails we built has it 🤷

Comment (MounirDhahri) [reply src/app/Scenes/HomeView/Sections/MarketingCollectionsRailHomeViewSection.tsx:19]: no need to address it in this PR of course

File: src/app/Scenes/HomeView/Sections/MarketingCollectionsRailHomeViewSection.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  const marketingCollections = extractNodes(data.marketingCollectionsConnection)
+  if (!marketingCollections || marketingCollections.length === 0) return null
+
+  return (
+    <Flex>
+      <Flex pl={2} pr={2}>
+        <SectionTitle title={component.title} />
+      </Flex>
+
+      <CardRailFlatList<any>
```
Comment (MounirDhahri) [inline src/app/Scenes/HomeView/Sections/MarketingCollectionsRailHomeViewSection.tsx:30]: fwiw, we have a node type helper that can be useful here
```suggestion
<CardRailFlatList<ExtractNodeType<MarketingCollectionsRailHomeViewSection_section["marketingCollectionsConnection"]>>
```

Comment (MounirDhahri) [reply src/app/Scenes/HomeView/Sections/MarketingCollectionsRailHomeViewSection.tsx:30]: Maybe we can export the type above the component

**Review by MounirDhahri — DISMISSED** [review]
Nice - you're on fire with these PRs 🔥

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ✅ **No changelog changes**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d847e678a0d878e207d5e9b931f5ff3f92e8b07c
</p>

**Pull request merged**